packages/identifier-resolution/src/types/IJwtService.d.ts

```diff
@@ -0,0 +1,226 @@
+export interface JweHeader extends Omit<BaseJwtHeader, 'alg'> {
+    alg: string;
+    enc: string;
+    jku?: string;
+    jwk?: BaseJWK;
+    epk?: EphemeralPublicKey;
+    x5u?: string;
+    x5c?: string[];
+    x5t?: string;
+    cty?: string;
+    crit?: string[];
+    [k: string]: any;
+}
+export interface JweRecipientUnprotectedHeader {
+    alg: string;
+    iv: string;
+    tag: string;
+    epk?: EphemeralPublicKey;
+    kid?: string;
+    apv?: string;
+    apu?: string;
+}
+export interface JweProtectedHeader extends Partial<JweHeader> {
+    zip?: 'DEF' | string;
+}
+export type Jws = JwsCompact | JwsJsonFlattened | JwsJsonGeneral;
+export type JwsCompact = string;
+export interface JwsJsonFlattened {
+    payload: string;
+    protected: string;
+    header?: JwsHeader;
+    signature: string;
+}
+export interface JwsJsonGeneral {
+    payload: string;
+    signatures: Array<JwsJsonSignature>;
+}
+export interface JwsJsonGeneralWithIdentifiers extends JwsJsonGeneral {
+    signatures: Array<JwsJsonSignatureWithIdentifier>;
+}
+export interface JwsJsonSignatureWithIdentifier extends JwsJsonSignature {
+    identifier: ExternalIdentifierResult;
+    publicKeyHex: string;
+}
+export type Jwe = JweCompact | JweJsonFlattened | JweJsonGeneral;
+export type JweCompact = string;
+export interface JweJsonFlattened {
+    protected: string;
+    unprotected: JweHeader;
+    header: JweHeader | JweRecipientUnprotectedHeader;
+    encrypted_key?: string;
+    aad?: string;
+    iv: string;
+    ciphertext: string;
+    tag?: string;
+}
+export interface JweRecipient {
+    header?: JweRecipientUnprotectedHeader;
+    encrypted_key?: string;
+}
+export interface JweJsonGeneral {
+    protected: string;
+    unprotected?: JweHeader;
+    recipients: Array<JweRecipient>;
+    aad?: string;
+    iv: string;
+    ciphertext: string;
+    tag?: string;
+}
+export interface PreparedJwsObject {
+    jws: PreparedJws;
+    b64: {
+        payload: string;
+        protectedHeader: string;
+    };
+    identifier: ManagedIdentifierResult;
+}
+export interface BaseJwtHeader {
+    typ?: string;
+    alg?: string;
+    kid?: string;
+}
+export interface BaseJwtPayload {
+    iss?: string;
+    sub?: string;
+    aud?: string[] | string;
+    exp?: number;
+    nbf?: number;
+    iat?: number;
+    jti?: string;
+}
+export interface JwsHeader extends BaseJwtHeader {
+    kid?: string;
+    jwk?: JWK;
+    x5c?: string[];
+    [key: string]: unknown;
+}
+export interface JwsPayload extends BaseJwtPayload {
+    [key: string]: unknown;
+}
+export interface JwsHeaderOpts {
+    alg: JoseSignatureAlgorithm | JoseSignatureAlgorithmString;
+}
+export type JwsIdentifierMode = 'x5c' | 'kid' | 'jwk' | 'did' | 'auto';
+export type EncryptJweCompactJwtArgs = {
+    payload: JwsPayload;
+    protectedHeader?: JweProtectedHeader | undefined;
+    aad?: Uint8Array | undefined;
+    recipientKey: ExternalIdentifierResult & {
```

Comment:
maybe nicer to make a type for this?

Reply:
I do not control that, this is a ".d.ts" typings file copied from jwt-service because I can't add that package to any dependency list